static/app/components/forms/formField/index.tsx

```diff
@@ -417,12 +417,13 @@ class FormField extends Component<FormFieldProps> {
                     type={saveMessageAlertType}
                     trailingItems={
                       <Fragment>
-                        <Button onClick={this.handleCancelField} size="xs">
+                        <Button onClick={this.handleCancelField} size="xs" type="button">
```

Comment:
In RTL this was submitting the form. I think something else was stopping this in actual production